elastic/kibana
# [Infra] Fix duplicate requests

fixes: [#189484](https://github.com/elastic/kibana/issues/189484)

## Summary

Fix duplicate request during hosts view mounting cycle.

https://github.com/user-attachments/assets/dce73b2a-77e8-461a-bd9e-661fc40afe7a


https://github.com/user-attachments/assets/0e8c41b4-1a36-470f-855d-c189221ad1f6


The problem happened because the `buildEsQuery` returns a complex object, making the `useFetcher` to treat every payload, as a new one, triggering duplicate requests.

### Extra

I have refactored the `use_unified_search` and `use_unified_search_url_state`. It was misusing `useReducer` and there was some magic happening with `rxJS`. I tried to make the code easier to understand

### How to test

- Start a local Kibana and es instances
- run `node scripts/synthtrace infra_hosts_with_apm_hosts --live `
- Navigate to Infrastructure > Hosts
  - Open the dev tools and check if there is more than one call to `api/metrics/infra` endpoint
